lib/record_store/provider/oracle_cloud_dns.rb

```diff
@@ -0,0 +1,215 @@
+          all_records += response.data.items
+        end
+
+        update_zone_record_items = all_records.map do |r|
+          OCI::Dns::Models::RecordDetails.new(
+            domain: r.domain,
+            ttl: r.ttl,
+            rtype: r.rtype,
+            recordHash: r.record_hash,
+            rdata: r.rdata,
+            rrsetVersion: r.rrset_version,
+            isProtected: true
+          )
+        end
+        update_zone_record_items << OCI::Dns::Models::RecordDetails.new(
+          domain: record_fqdn,
+          ttl: record.ttl,
+          rtype: record.type,
+          rdata: record_rdata
+        )
+        update_zone_record_items.delete_if { |r| id == r.record_hash }
+
+        client.update_zone_records(
+          zone,
+          OCI::Dns::Models::UpdateZoneRecordsDetails.new(items: update_zone_record_items)
+        )
+      end
+
+      def secrets
+        super.fetch('oracle_cloud_dns')
+      end
+
+      def build_from_api(api_record)
+        fqdn = Record.ensure_ends_with_dot(api_record.domain)
+
+        record_type = api_record.rtype
+        return if record_type == 'SOA'
+
+        record = {
+          ttl: api_record.ttl,
+          fqdn: fqdn.downcase,
+          record_id: api_record.record_hash,
+        }
+        case record_type
+        when 'A', 'AAAA'
+          record.merge!(address: api_record.rdata)
+        when 'ALIAS'
+          record.merge!(alias: api_record.rdata)
+        when 'CAA'
+          flags, tag, value = api_record.rdata.split(' ')
+
+          record.merge!(
+            flags: flags.to_i,
+            tag: tag,
+            value: Record.unquote(value),
+          )
+        when 'CNAME'
+          record.merge!(cname: api_record.rdata)
+        when 'MX'
+          preference, exchange = api_record.rdata.split(' ')
+          record.merge!(
+            preference: preference.to_i,
+            exchange: exchange,
+          )
+        when 'NS'
+          record.merge!(nsdname: api_record.rdata)
+        when 'SPF', 'TXT'
+          record.merge!(txtdata: Record.unquote(api_record.rdata).gsub("\"", ""))
+        when 'SRV'
+          priority, weight, port, host = api_record.rdata.split(' ')
+          record.merge!(
+            priority: priority.to_i,
+            weight: weight.to_i,
+            port: port.to_i,
+            target: Record.ensure_ends_with_dot(host),
+          )
+        end
+        Record.const_get(record_type).new(record)
+      end
+
+      def retrieve_rdata_based_on_record_type(record)
+        case record.type
+        when 'ALIAS'
+          record.rdata[:alias]
+        when 'A'
+          record.rdata[:address]
+        when 'CAA'
+          [record.flags.to_s, record.tag, record.value].join(' ')
+        when 'SRV'
+          [record.priority.to_s, record.weight.to_s, record.port.to_s, record.target].join(' ')
+        when 'CNAME'
+          record.cname
+        when 'MX'
+          [record.preference, record.exchange.to_s].join(' ')
```

Comment:
Interpolation is more difficult to maintain (and less importantly, slower).

Reply:
Will stick with Chris's comment,

Reply:
> Interpolation is more difficult to maintain (and less importantly, slower).

I wanted to fact-check things about that, so I went and found pro and cons about interpolation vs. concatenation (which is what `join` is doing): https://stackoverflow.com/questions/10076579/string-concatenation-vs-interpolation-in-ruby
Which includes, among others, as pro for `interpolation`, that it is _More idiomatic within the Ruby community_ This was one of my point here: interpolation in the case where you want to return a string is what should be done, because it's clear about your return intention. The `join` with a number of `to_s` makes things harder to read and understand, and thus *harder* to maintain.

Also, we can see in the pros that `interpolation` is _Faster to accomplish during runtime_. So seemed like @cttttt and `Phrogz` weren't agreeing on that. So I took none of your works and made a small comparison script: 😝 

```ruby
require 'benchmark'

integer = 2
value = "something"

puts "Interpolation:"
puts Benchmark.measure {
  50_000.times do
    "#{integer} #{value}"
  end
}

puts "Join:"
puts Benchmark.measure {
  50_000.times do
    [integer.to_s, value].join(' ')
  end
}
```

And here are the results:
```
$ruby main.rb
Interpolation:
  0.040000   0.000000   0.040000 (  0.035645)
Join:
  0.060000   0.000000   0.060000 (  0.067454)
```

And another run because I did not trust the first set of results:
```
$ruby main.rb
Interpolation:
  0.040000   0.000000   0.040000 (  0.067609)
Join:
  0.070000   0.000000   0.070000 (  0.133417)
```

And finally while reverting the order of the tests, just to be sure:
```
$ruby main.rb
Join:
  0.060000   0.000000   0.060000 (  0.074109)
Interpolation:
  0.030000   0.000000   0.030000 (  0.072111)
```

So... Given those results and the question of maintainability, conclusions are:
- ~more difficult to maintain~
- ~slower~

I would thus stand by the move to interpolation 😉 - But again, you can ignore me hehe, I don't mind 🤷‍♂

Reply:
Interesting.  Interpolation performs almost as good as straight concatenation.  Way to use data to get to the truth @XaF !